iceoryx2-cal/src/dynamic_storage/posix_shared_memory.rs

```diff
@@ -136,45 +200,108 @@ impl<T: Send + Sync + Debug> NamedConceptBuilder<Storage<T>> for Builder<T> {
-        self
-    }
+impl<'builder, T: Send + Sync + Debug> Builder<'builder, T> {
+    fn open_impl(&self) -> Result<Storage<T>, DynamicStorageOpenError> {
+        let msg = "Failed to open ";
 
-    fn supplementary_size(mut self, value: usize) -> Self {
-        self.supplementary_size = value;
-        self
+        let full_name = self.config.path_for(&self.storage_name).file_name();
+        let mut adaptive_wait = fail!(from self, when AdaptiveWaitBuilder::new().create(),
+                                    with DynamicStorageOpenError::InternalError,
+                                    "{} since the AdaptiveWait could not be initialized.", msg);
+
+        let mut elapsed_time = Duration::ZERO;
+        let shm = loop {
+            match SharedMemoryBuilder::new(&full_name).open_existing(AccessMode::ReadWrite) {
+                Ok(v) => break v,
+                Err(SharedMemoryCreationError::DoesNotExist) => {
+                    fail!(from self, with DynamicStorageOpenError::DoesNotExist,
+                    "{} since a shared memory with that name does not exists.", msg);
+                }
+                Err(SharedMemoryCreationError::InsufficientPermissions) => {
+                    if elapsed_time >= self.timeout {
+                        fail!(from self, with DynamicStorageOpenError::InitializationNotYetFinalized,
+                        "{} since it is not yet readable - most likely since it is not finalized after {:?}.",
+                        msg, self.timeout);
+                    }
+                }
+                Err(_) => {
+                    fail!(from self, with DynamicStorageOpenError::InternalError, "{} since the underlying shared memory could not be opened.", msg);
+                }
+            };
+
+            elapsed_time = fail!(from self, when adaptive_wait.wait(),
+                                    with DynamicStorageOpenError::InternalError,
+                                    "{} since the adaptive wait call failed.", msg);
+        };
+
+        let required_size = std::mem::size_of::<Data<T>>() + self.supplementary_size;
+        if shm.size() < required_size {
+            fail!(from self, with DynamicStorageOpenError::InternalError,
+                "{} since the actual size {} does not match the required size of {}.", msg, shm.size(), required_size);
+        }
+
+        let init_state = shm.base_address().as_ptr() as *const Data<T>;
+
+        // The mem-sync is actually not required since an uninitialized dynamic storage has
+        // only write permissions and can be therefore not consumed.
+        // This is only for the case that this strategy fails on an obscure POSIX platform.
+        //
+        //////////////////////////////////////////
+        // SYNC POINT: read Data<T>::data
+        //////////////////////////////////////////
+        let package_version = unsafe { &(*init_state) }
+            .version
+            .load(std::sync::atomic::Ordering::Acquire);
```

Comment:
Since this is not in the hot path, what do you think about using `SeqCst`?

Reply:
Could you explain why? Just be more on the safer side?

Reply:
This - you guy - seems like a smarty one ... I will listen to him.